websites/M/Mixcloud/presence.ts

```diff
@@ -4,72 +4,72 @@ const presence = new Presence({
   strings = presence.getStrings({
     play: "presence.playback.playing",
     pause: "presence.playback.paused",
-    live: "presence.activity.live"
+    live: "presence.activity.live",
   });
 
-let live, prevLive, author: string, title: string;
+let author: string, title: string, url: string;
 
 presence.on("UpdateData", async () => {
   const player = document.querySelector(
-    ".PlayerControls__PlayerContainer-vo7mt3-4"
+    "[class^='PlayerControls__PlayerContainer']"
   );
 
   if (player) {
-    const paused =
-        document.querySelector(".PlayButton__PlayerControl-rvh8d9-1") === null,
-      on_air = document.querySelector(
-        ".LiveStreamPage__LiveStreamPageHeader-hwzri8-1"
-      );
+    const normalIsPlaying: boolean =
+      document
+        .querySelector("div[class^='PlayButton__PlayerControl']")
+        ?.getAttribute("aria-label") === "Pause";
 
-    if (on_air) {
-      live = true;
-      if (prevLive !== live) {
-        prevLive = live;
-      }
-    } else {
-      live = false;
-    }
+    const liveIsPlaying: boolean =
+      document
+        .querySelector("[class^=LiveVideo__VideoContainer] .shaka-play-button")
+        ?.getAttribute("icon") === "pause";
+
+    const isPlaying = normalIsPlaying || liveIsPlaying;
+
+    if (normalIsPlaying) {
+      const normalDetails = document.querySelector(
+        "[class^='shared__ShowDetails'] > a:nth-child(1)"
+      );
 
-    if (!live) {
+      title = normalDetails.textContent;
+      url = new URL(normalDetails.getAttribute("href"), window.location.origin)
+        .href;
+      author = document.querySelector(
+        "[class^='PlayerControls__ShowOwnerName']"
+      ).textContent;
+    } else if (liveIsPlaying) {
+      url = window.location.href;
       title = document.querySelector(
-        ".shared__ShowDetails-gqlx6k-1 > a:nth-child(1)"
+        "[class^='LiveStreamDetails__StreamTitleContainer'] > h4"
       ).textContent;
       author = document.querySelector(
-        "p.Typography__BodyExtraSmall-sc-1swbs0s-16:nth-child(2) > a:nth-child(1)"
+        "[class^='LiveStreamStreamerDetails__StreamerDetailsTextContainer'] h4"
       ).textContent;
-    } else {
-      title = document.querySelector(".Typography__Headline1Small-sc-1swbs0s-8")
-        .textContent;
-      author = "Mixcloud Live";
     }
 
     const data: PresenceData = {
       details: title,
       state: author,
       largeImageKey: "mixcloud",
-      smallImageKey: paused ? "pause" : "play",
-      smallImageText: paused ? (await strings).pause : (await strings).play
+      smallImageKey: isPlaying ? "play" : "pause",
+      smallImageText: isPlaying ? (await strings).play : (await strings).pause,
+      buttons: [{ label: 'OPEN', url: url }]
```

Comment:
what is this button for bc i dont get it?
![image](https://user-images.githubusercontent.com/9348108/113506323-76e58480-9544-11eb-918c-aa75aa2d8ce2.png)

Reply:
the button should open the current music the user is listening. Either the normal played music or the live stream.

Reply:
I can remve the button, it seems not to work anyways.. No button works for me, for example the youtube presence, shows 2 buttons but nothing happens if I click it, no idea why. Maybe this is just me and it works for everyone else. (using the windows discord app client)

Reply:
Yeah, it's discord thing. They work but they don't work for yourself. I mean, okay then? Could you show me example where it will redirect? by logging or so?

Reply:
I added a console.log when it is playing for the target url, hope a video is ok? if not I can add console.log and push and you can try by yourself (I think Codacy / ESlint might have a problem with console.log then because of "no-console")

https://ttprivatenew.s3.amazonaws.com/pulse/pits-michel/attachments/16444882/TinyTake04-04-2021-02-12-22.mp4

so the url is basically the url of the played stream. If a user opens the urk, he will land on the stream page where the stream can be played / commented etc.. or in the case of a live stream the stream url will be opened with chat etc.

Reply:
Okay cool, could you change text to like idk `View Song` or something? And will be good to accept :D

Reply:
I just pushed a change. The button will change based on what is played. If a user is watching a live stream the button label is "View Livestream". And if the user is only listening to normal music it will show "Listen to Show". Mixcloud calls these non live recorded streams "Shows" not just song. It only allows longer music streams / mixes. So I named the button also "Shows". Hope it is ok. Let me know if there is still stomething I should do.